### PR TITLE
260 conditional pipeline parts load

### LINKED ISSUE #260: Conditional pipeline parts load
Introduce `condition`-based blocks like:

```yaml
pipeline:
  elements:
    init_condition: 
      expr: expression-on-config-vars
      value: "something"

    - element: nvinfer@detector
    - element: ...
  elements:
    ...
    ...
  
```

The behavior is static, not dynamic. Multiple `elements` blocks may present.

### PR DESCRIPTION
**Pull request by abramov-oleg**

### CODE REVIEW

File: samples/traffic_meter/README.md [hunk 1]
```diff
@@ -87,22 +87,24 @@ Now you are ready to run the performance benchmark with the following command:
 
 ```bash
 docker run --rm -it --gpus=all \
+-e DETECTOR=peoplenet \
```
Comment (tomskikh) [inline samples/traffic_meter/README.md:97]: Add an instruction for choosing between peoplenet or yolov8m detector for demo.

File: savant/config/schema.py [hunk 2]
```diff
@@ -352,8 +364,14 @@ class Pipeline:
                 element: uridecodebin
                 properties:
                     uri: file:///data/test.mp4
-            elements:
-                # user-defined pipeline elements
+            stages:
```
Comment (tomskikh) [inline savant/config/schema.py]: Update example

File: savant/config/module_config.py [hunk 3]
```diff
@@ -152,34 +173,127 @@ def setup_batch_size(config: Module) -> None:
...
 
     :param config: Module config.
     """
+    if 'parameters' not in module_cfg or module_cfg.parameters is None:
+        module_cfg.parameters = {}
+        return
+
+    def apply_schema(cfg: dict, node: str, schema_class: Any) -> None:
+        schema = OmegaConf.structured(schema_class)
+        if node not in cfg or cfg[node] is None:
```
Comment (tomskikh) [inline savant/config/module_config.py:191]: `draw_func` is optional. When it's `None` or absent it should stay that way. I suggest to add `default` parameter to this method.

Here `draw_func` is always enabled.

File: docs/source/savant_101/54_conditional_unit_init.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  :language: YAML
+  :lines: 27-45
+
+Later in the module config another group is defined for the yolov8m detector:
+
+.. literalinclude:: ../../../samples/traffic_meter/module.yml
+  :language: YAML
+  :lines: 67-75
+
+The groups are initialized based on whether their :py:attr:`~savant.config.schema.ElementGroup.init_condition` evaluates to `True` at the time of module initialization. In the example above which detector unit is initialized depends on the value of the ``DETECTOR`` environment variable. If it is set to `peoplenet` then the peoplenet detector is initialized, and similarly for the yolov8m detector.
```
Comment (tomskikh) [inline docs/source/savant_101/54_conditional_unit_init.rst:18]: Describe what syntax is used to evaluate init_condition expression

**Review by tomskikh — CHANGES_REQUESTED** [review]

File: samples/traffic_meter/README.md (hunk 1 above)
Comment (abramov-oleg) [reply samples/traffic_meter/README.md:97]: Extended the README with the model choice explanation.

File: savant/config/schema.py (hunk 2 above)
Comment (abramov-oleg) [reply savant/config/schema.py]: Fixed the docstring.

File: savant/config/module_config.py (hunk 3 above)
Comment (abramov-oleg) [reply savant/config/module_config.py:191]: Fixed.

File: docs/source/savant_101/54_conditional_unit_init.rst (hunk 4 above)
Comment (abramov-oleg) [reply docs/source/savant_101/54_conditional_unit_init.rst:18]: Added an explanation.

**Review by tomskikh — APPROVED** [review]

**Pull request merged**